### DISCUSSION TITLE
Github Actions Logs in Light Mode Looks Like Dark Mode

### DISCUSSION THREAD

**Discussion by fw-aaron** [community/community#76350] [Accessibility] [answered]
### Select Topic Area

Bug

### Body

https://github.com/orgs/community/discussions/22208 shows the basic state of the situation. Using any light mode theme, the logs have a dark background and light text which is antithetical to light mode. When I click "view raw logs", the browser shows a light background and dark text which is better, but it would be nice if I could see them properly without having to resort to raw logs.

**Comment by ghostinhershell** [comment]
Hi there @fw-aaron, 

Thanks so much for reaching out about this. I see that this is an older conversation, so I'll follow-up with the team to check on the status of this particular request.

**Comment by cehfisher** [comment]
Hi @fw-aaron - I believe this color contrast issue was resolved. Please review [this comment](https://github.com/orgs/community/discussions/22208#discussioncomment-9498936) from another issue that is a duplicate. Thanks! (cc @queenofcorgis)

**Reply by fw-aaron** [reply]
Thank you for contacting me back! The issue is not resolved as the light mode is still ignored in the "main" area of Microsoft Github Actions window. See this image:

![image](https://github.com/community/community/assets/57676146/b0fa6772-3fb6-42cc-886a-b8e68cc464b9)

The red arrows point to the area I'm talking about. Maybe I was unclear in the original comment, but hopefully this image clears things up. The text in that area (what I might call the "main" area of this page) is "white on black" which would be appropriate for "dark mode", but doesn't match a "light mode" them. I can't find any way to make that black box match its surroundings.

**Comment by cehfisher** [comment]
Thank you @fw-aaron for letting us know that this is a different issue than the one that was resolved. I will follow-up internally with the accessibility team on next steps. We really appreciate the time you took to give us feedback!

**Reply by sitase** [reply]
Month and a half later, do you have any feedback from the a11y team?

**Reply by cehfisher** [reply]
Hello @sitase - the issue has been reported internally and work is in progress. Theme changes can take some time. Thank you for your patience.

**Reply by fw-aaron** [reply]
> ...work is in progress. Theme changes can take some time.

That's great! Is there a concrete plan already? Is light mode going to keep the light background all over the page? Is dark mode going to have a dark background all over the page? Maybe some other solution I can't currently imagine? Thanks a ton!

**Comment by lakesare** [comment]
@cehfisher, I'd like to note that this is not merely a UI issue, this is an accessibility issue.

I have a hard time actually *seeing* anything in the github actions pane because it's white-on-black text.
I have to copy&paste results of the action log to be able to read it.

**Comment by pascal-hofmann** [comment]
Hi,
I also have the same issue. My eyes have a hard time adjusting to the mixed light/dark themed page and in general I have issues reading things in dark mode.

It would be great to get this fixed.

**Comment by zlobober** [comment]
Hi,
Recently I started using light themes because of deterioration of vision, and the current state of the action page in the light scheme is really annoying. Having a mixture of massive amount of dark-themed information on a light-themed page is even worse than either of consistent dark and light themes.
I'd highly appreciate if this issue was addressed soon.

**Reply by fw-aaron** [reply]
My situation is different. I love "dark" theme, however lots of products and web pages don't support a dark theme, and of those that do, 1) it takes some time to set it up on each computer I use or create, and 2) it often works imperfectly. Nearly everything supports light theme, so I invert the colors on my monitor to make sure everything is good. I can change the inversion with a key press so I can see pictures and videos (although I'm getting better at seeing them inverted without the need to un-invert ^_^). Obviously having a page that is half light theme and half dark theme breaks everyone's preference.

**Comment by noah56** [comment]
This has bothered me enough that I've found a solution for myself until this gets actually fixed by Github. 
I'm using Arc browser, which has a boost feature to allow users to modify styling of webpages with CSS and JS. I created a boost with the following JS:

```
document.addEventListener('DOMContentLoaded', function() {
    const element = document.querySelector('#logs');
    if (element) {
        element.setAttribute('data-color-mode', 'light');
    }
});
```

A message to the GitHub devs: Please allow the data-color-mode of logs to respect the user's light/dark mode preferences.

For Chrome/Firefox users, I'm sure there's a way to run JS on page load, perhaps with an extension.
See screenshot for new styling. Quite nice IMO:
<img width="1268" alt="image" src="https://github.com/user-attachments/assets/f4e926d6-c10e-4f90-99ea-9c12b2906870">

**Reply by lakesare** [reply]
Oh that's quite exceptional. I expected this would warrant a ton of css changes instead of a simple toggle (this toggle does work for me). Thank you for looking into a fix!

**Reply by limitedAtonement** [reply]
I really should write my own browser extension that is sort of a catch-all for the various annoyances of sites with which I interact. @noah56 I owe you a beer or two.

**Reply by jvincent-siplay** [reply]
Adding that script to Greasemonkey in Firefox works great, thank you!!

I also added this custom CSS (using the Stylus extension) to fix the white text on light grey (visible in that screenshot above the line "succeeded 10 hours ago"):

```
#logs[data-color-mode="light"] .color-text-white {
    color: var(--fgColor-default) !important;
}
```

**Reply by OothecaPickle** [reply]
here's a userscript that changes the theme of the log area and the color of the job title text

```js
// ==UserScript==
// @name         GitHub Actions Log White Theme Fix
// @version      1.0
// @description  fix color of Actions log when using white theme
// @match        https://github.com/*/*/actions/runs/*
// @grant        GM_addStyle
// ==/UserScript==

(function() {
    const element = document.querySelector('#logs')
    if (element) {
        element.setAttribute('data-color-mode', 'light')
    }
    GM_addStyle ( `
        #logs[data-color-mode="light"] .color-text-white {
            color: var(--fgColor-default) !important;
        }
    ` )
})()
```

**Comment by kobenguyent** [comment]
another approach is using chrome devtools snippets

![Snippets](https://github.com/user-attachments/assets/2b8eb5a7-85fa-4531-ac12-e40b169b57ab)

```
// Select the target element
const logsElement = document.getElementById("logs");

// Check if the element exists
if (logsElement) {
    // Change `data-color-mode` to 'light'
    logsElement.setAttribute("data-color-mode", "light");

    // Optional: Change any other styles or attributes if necessary
    logsElement.style.backgroundColor = "#ffffff"; // Set a light background color
    logsElement.style.color = "#000000"; // Set a dark text color
    
    console.log("Color mode changed to light.");
} else {
    console.error("Element with id 'logs' not found.");
}
```

![light theme](https://github.com/user-attachments/assets/669ff575-9271-498e-9a32-55635a3d528a)

**Comment by phalicz-apptio** [comment]
GitHub team could you please officially fix it? Given the fix suggestions mentioned above it looks like a low hanging fruit.

It's a real accessibility issue for people who have trouble seeing white text on black background. Not everyone can install browser extensions to workaround it.

**Reply by cehfisher** [reply]
Hey @phalicz-apptio – the internal team is aware of this issue, but it looks like it’s a bit more complex to resolve. We’ll share updates as soon as we have more info. Thanks for your continued patience!

**Comment by timh1975** [comment]
I would find having light mode for GItHub actions highly beneficial as well. I've created a GitHub action as part of a undergaduate software engineering degree. Copying snippets in dark mode to a report makes it much harder to read.

**Comment by pfox-hpc** [comment]
Another 'me too' here. Please fix, I can't stand dark mode, very hard to read. Thanks!

**Comment by noah56** [comment]
Still not fixed... and I've stopped using Arc browser, so I went ahead and created a Chrome extension to solve this problem if anyone prefers a one click install solution. I've included the fixes mentioned above, setting `data-color-mode` to `light` and setting `.color-text-white` to `--fgColor-default`. Just modified a bit to work as a chrome extension.

It runs on the url pattern `https://github.com/*/*/actions/runs/*` on `document_idle`

[Chrome web store link
](https://chromewebstore.google.com/detail/gmohjegpboaejcpdopfjocieidbadhnn?utm_source=item-share-cb)

[Github repo](https://github.com/noah56/github-actions-light-mode)

**Comment by cehfisher** [comment, marked as answer]
Hi @fw-aaron and everyone commenting/following this post – good news! The Actions log viewer now correctly respects the user's selected theme preference across all GitHub themes, including those designed to support users with color deficiency issues 🎉

Please help us retest this update by clearing your browser cache and giving it another try. If you’re still encountering issues, feel free to drop a comment here. Thanks again for your patience!

**Reply by sebthom** [reply]
Can this be made optional/reverted? I prefer reading terminal style logs on black background and sourcecode on white. Also since now the background is light gray and the text is dark gray it lost contrast and feels tiring to my eyes.

**Reply by cehfisher** [reply]
Thanks for your feedback, @sebthom! This update fixes a bug where the Actions log viewer wasn’t respecting users’ theme preferences, so there isn’t an option to toggle this behavior on or off.

If you prefer a dark background, you can choose from four different dark themes and test out each one to find the best fit for you. Just go to Profile → Appearance → Theme preferences (https://github.com/settings/appearance). Thanks!

**Reply by peterpakos** [reply]
I normally use light theme in OS/apps but I much prefer having dark Actions log viewer background, the current light background just hurts my eyes! I don't really want to change to dark theme to get the old logs background back. If this cannot be reverted/made optional, can dark log viewer background be made optional on light theme?

**Reply by leemeador** [reply]
I don't want dark everywhere. I want the workflow logs to have the dark background.

**Reply by cehfisher** [reply]
Thanks for the feedback, @peterpakos! We've shared your and @sebthom’s comments with the team that implemented the fix. If either of you could open a new discussion thread, that would be helpful, since this one will close once the original poster confirms the issue is resolved. Since this report dates back to 2023, starting a new thread helps keep everything on track and ensures feedback stays organized.

**Reply by limitedAtonement** [reply]
Looks good, thank you.

![image](https://github.com/user-attachments/assets/5c39671e-05f9-4c49-90d0-e5b6af747261)

**Reply by fw-aaron** [reply]
Great. Thank you, Carrie. This looks right to me.

To those who don't like the more homogenous coloring scheme, you can set `xrandr-invert-colors` to a shortcut key in your window manager, or if you use Microsoft Windows, try https://zerowidthjoiner.net/negativescreen.

I can't imagine wanting some text to be white on black and some black on white in the same screen...either way, you'll have some portion of the screen scorching into your skull and the other part of your screen understated. If everything on the screen is either light or dark background, you can invert it to what you like using the tools mentioned above. Good luck out there!

**Reply by sebthom** [reply]
For anyone interested I created a CSS patch that can be applied with [Stylus](https://github.com/openstyles/stylus/) to revert this change. Might not be perfect but gets the job done for me https://gist.github.com/sebthom/2564ef486a913ed1b60e9dfb6ec6568f

**Comment by Callek** [comment]
Just wanted to come here and say "I see this fixed" -- however I'm a user with the opposite preference.

* I prefer light mode in general
* I really prefer my Console-Like logs in dark mode

This breaks my ease-of-reading for several CLI tools with color output in my github logs, since those tools expect a black (or at least dark) background.

I would enjoy a separate configuration option to allow me to keep all of github as light mode, but force dark mode for this.

Thank you for the consideration.

**Reply by cehfisher** [reply]
Thanks @Callek for sharing your perspective! As mentioned above, if you’re able to open a new discussion thread, that would be really helpful as this one will close once the original poster confirms the original issue is resolved. Starting a new thread helps us preserve feedback that might otherwise be lost. We'll also make sure to add your input to the team’s feedback on this bug fix!

**Reply by fw-aaron** [reply]
> ...CLI tools with color output...

That's an important consideration! If your Actions output has light colors, you can use dark mode to make the background dark, then the light-colored text will be easy to read again.

**Reply by Callek** [reply]
> > ...CLI tools with color output...
> 
> That's an important consideration! If your Actions output has light colors, you can use dark mode to make the background dark, then the light-colored text will be easy to read again.

That is certainly an option, but flies in the face of my general preference for websites and code to be in a light background

**Reply by sebthom** [reply]
@Callek I am totally with you. As a quick workaround I am now applying this CSS using [stylus](https://github.com/openstyles/stylus/) which mostly reverts the change:
```css
#logs * {
  color: white;
  background-color: #24292E;
}
#logs svg.octicon-check-circle-fill {
  fill: #1A7F37 !important;
}
#logs svg.octicon-x-circle-fill {
  fill: #D1242F !important;
}
#logs .log-line-warning, #logs .log-line-warning *  {
  background-color:#44402A;
}
#logs .color-fg-attention {
  color: #FFEA7F !important;
}
#logs .log-line-error, #logs .log-line-error *  {
  background-color:#3D282E;
}
```

I also created a gist for it https://gist.github.com/sebthom/2564ef486a913ed1b60e9dfb6ec6568f

**Reply by limitedAtonement** [reply]
> ... all of github as light mode, but force dark mode for this.

> ...my general preference for websites and code to be in a light background

I'm not suggesting that you permanently invert everything. I'm thinking that when you're looking at the Microsoft Github Actions window, press <kbd>Super</kbd>+<kbd>m</kbd> (or whatever your shortcut is), then when you're done, press it again to un-invert. You may find it convenient to invert anything you don't like. Personally, I use light mode (because there is almost always a light mode by default), then invert everything so that I'm basically in a complete dark mode with no exceptions because I work in a dark room where bright lights are unwelcome, and I don't have to configure anything. If I watch videos or need to see details on pictures, I can uninvert temporarily.

**Comment by Callek** [comment]
As requested upthread, I have created a new discussion for the UX breakage fixing this created.

https://github.com/orgs/community/discussions/163594

**Reply by cehfisher** [reply]
Thank you so much @Callek! We will close this issue out at EOD if the original poster @fw-aaron doesn't close it before then 😸

**Comment by Vinccool96** [comment]
Damn, why did you guys fix it :(

**Comment by NGPixel** [comment]
Why was this not rolled out behind an option in the user preferences? Like many users, I want the site in light mode but the console in dark mode.
